### DISCUSSION TITLE
DebugSimpleRender question.

### DISCUSSION THREAD

**Discussion by benjaminleezhiyuan** [jrouwe/JoltPhysics#1361] [Q&A]
![image](https://github.com/user-attachments/assets/878368bd-fe76-4e73-84a1-0d9a9168eba0)
Currently when I have a single body created I am able to use my DrawLine function to draw the body of the object.

![image](https://github.com/user-attachments/assets/d720e806-a86d-4688-9d5f-750af8f42ea6)
But when I added the body above it looks like this. Sometimes nothing would be drawn. Is it a limitation of the simple renderer?

Another question is SimpleRender only able to draw boxes or can they draw all the other shapes as well?

**Comment by jrouwe** [comment]
The DebugRendererSimple should be able to render any shape and I'm not aware of any limitations.

**Reply by benjaminleezhiyuan** [reply]
So as long as there are the proper drawline functions etc any bodies that are added to the system will be drawn using the drawbodies?

**Reply by jrouwe** [reply]
Yes, that should be the case (bugs are always possible ofc).

**Reply by benjaminleezhiyuan** [reply]
So I inherited from the simple class
Then I constructed the renderer
I set the settings
Then call drawbodies(settings,renderer) in my main loop.

**Reply by jrouwe** [reply]
That sounds correct. I suggest stepping into the draw code in the debugger. DrawBodies is a very simple loop and the DebugRendererSimple is also very simple.
